geniza/corpus/tests/test_sitemaps.py

```diff
@@ -0,0 +1,81 @@
+from datetime import date
+from unittest.mock import patch
+
+from django.urls import reverse
+from parasolr.django import SolrClient
+
+from geniza.corpus.models import Document
+from geniza.corpus.sitemaps import (
+    DocumentScholarshipSitemap,
+    DocumentSitemap,
+    solr_timestamp_to_date,
+)
+from geniza.corpus.tests.conftest import make_document
+from geniza.footnotes.models import Footnote
+
+
+def test_solr_timestamp_to_date():
+    assert solr_timestamp_to_date("2020-05-12T15:46:20.341Z") == date(2020, 5, 12)
+
+
+class TestDocumentSitemap:
+    def test_items(self, document, suppressed_document):
+        # Ensure that documents are supressed if they aren't public
+        assert document.status == Document.PUBLIC
+        SolrClient().update.index(
+            [
+                document.index_data(),  # no scholarship records
+                suppressed_document.index_data(),  # suppressed
+            ],
+            commit=True,
+        )
+        sitemap = DocumentSitemap()
+        assert document.id in [obj["pgpid"] for obj in sitemap.items()]
+        assert suppressed_document.id not in [obj["pgpid"] for obj in sitemap.items()]
+
+    def test_location(self):
+        assert DocumentSitemap().location(
+            {"pgpid": "444", "last_modified": "2020-05-12T15:46:20.341Z"}
+        ) == reverse("corpus:document", args=["444"])
+
+    def test_lastmod(self):
+        assert DocumentSitemap().lastmod(
+            {"last_modified": "2020-05-12T15:46:20.341Z"}
+        ) == date(2020, 5, 12)
+
+
+class TestDocumentScholarshipSitemap:
+    def test_items(self, document, source, suppressed_document, fragment):
+        document_with_footnote = make_document(fragment, id=41)
```

Comment:
@rlskoeser 
I've been reading up on django testing, and structuring our tests with object factories seems to make the most sense because of all their dependencies. Have we tried [Factory Boy](https://factoryboy.readthedocs.io/en/stable/orms.html) before? You set defaults and then overwrite them for your specific use case. I know it's a bit late to introduce something like this, but I feel like half our tests are building objects.

Reply:
@kmcelwee I'm not familiar with this library, but I don't want to change approach right now, and don't want to introduce multiple new dependencies (I see there is a separate pytest plugin to connect pytest with factory boy).